sklearn/preprocessing/imputation.py

```diff
@@ -374,3 +376,368 @@ def transform(self, X):
             X[coordinates] = values
 
         return X
+
+
+class KNNImputer(BaseEstimator, TransformerMixin):
+    """Imputation for completing missing values using k-Nearest Neighbors.
+
+    Each sample's missing values are imputed from up to ``max_neighbors``
+    nearest neighbors found in the training set. Each missing feature is then
+    imputed as the average, either weighted or unweighted, of these neighbors
+    who have a value for it. Where all neighbors have that feature value
+    missing, the training set average for that feature is used for imputation.
+
+    Parameters
+    ----------
+    missing_values : integer or "NaN", optional (default = "NaN")
+        The placeholder for the missing values. All occurrences of
+        `missing_values` will be imputed. For missing values encoded as np.nan,
+        use the string value "NaN".
+
+    max_neighbors : int, optional (default = 5)
```

Comment:
This means we're not implementing the reference  paper, if I understand it correctly. It says "this  method  would
find  K  other genes, which have a value present in experiment 1". Is that what fancyimpute and the R packages do?

Reply:
Yes, we are following the R reference package and the paper